pkg/profiler/profile_writer.go

```diff
@@ -59,29 +60,35 @@ func (fw *FileProfileWriter) Write(_ context.Context, labels model.LabelSet, pro
 // RemoteProfileWriter is a profile writer that writes profiles to a remote profile store.
 type RemoteProfileWriter struct {
 	profileStoreClient profilestorepb.ProfileStoreServiceClient
+	// pool of gzip encoders helps to reduce GC pressure.
+	pool sync.Pool
 }
 
 // NewRemoteProfileWriter creates a new RemoteProfileWriter.
 func NewRemoteProfileWriter(profileStoreClient profilestorepb.ProfileStoreServiceClient) *RemoteProfileWriter {
 	return &RemoteProfileWriter{
 		profileStoreClient: profileStoreClient,
+		pool: sync.Pool{New: func() interface{} {
+			z, _ := gzip.NewWriterLevel(nil, gzip.BestSpeed)
```

Comment:
Let's not ignore errors :)

Reply:
It's safe enough because `gzip.NewWriter()` does the same thing https://github.com/klauspost/compress/blob/master/gzip/gzip.go#L58. There will be no error since `gzip.BestSpeed` is a constant.

Reply:
I hear you, however, the underlying implementation can change, and I don't feel comfortable ignoring errors. We can change the API and return the error to the caller and let it handle it.

Reply:
Sounds good.

Reply:
Oh, `sync.Pool.New` accepts `func() interface{}`, so in order to return an error along with `*gzip.Writer` I would have to introduce a struct to wrap those two arguments.

Reply:
If there is an error, then a pool would return `nil` and type casting will panic `rw.pool.Get().(*gzip.Writer)`, so the error will be spotted immediately. Additionally, we can pass a logger to `NewRemoteProfileWriter()` and log an error so it will be easier to figure out why `sync.Pool.New` returned a nil.